src/monodroid/jni/monodroid-glue.cc

```diff
@@ -2327,7 +2329,7 @@ MonodroidRuntime::Java_mono_android_Runtime_register (JNIEnv *env, jstring manag
 
 	MonoMethod *register_jni_natives = registerType;
 #if !defined (NET6)
-	MonoDomain *domain = mono_domain_get ();
+	MonoDomain *domain = utils.get_current_domain (/* attach_thread_if_needed */ false);
```

Comment:
How does `mono_thread_attach()` differ from `mono_jit_thread_attach()`?  Why use `utils.get_current_domain (/* attach_thread_if_needed */ false)` instead of using `utils.get_current_domain ()` and removing the `mono_jit_thread_attach()` invocation?

Reply:
`mono_jit_thread_attach` will call `mono_get_root_domain()` when the `domain` passed to it is `NULL`, basically what our `utils.get_current_domain()` does and it also switches to the new domain if it's different than the current one (something `mono_thread_attach` doesn't do). The reason I left it around and added the call to `utils.get_current_domain` is because I usually try not to make as few changes to existing code as possible, using `utils.get_current_domain` is consistent throughout the code and there's the subtle side effect of `mono_jit_thread_attach` switching to the passed domain.